### DISCUSSION TITLE
Send websocket text to client from another Function.

### DISCUSSION THREAD

**Discussion by Mike0675** [cesanta/mongoose#1971] [Ask questions, get answers] [answered]
Excuse this probably dumb Question, Im Hobbyist and learning C bit by bit.

Im testet the WS-Server Tutorial and it works.

Now i want to send Websocket Text from another function (for example  sendText() ).
Hows to do that?

When i do:

mg_ws_send(c, "test",5, WEBSOCKET_OP_TEXT);

my Compiler claims "Symbol 'c' could not be resolved".

So I think I have to give the function access to struct mg_connection *c which is in 
static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data).

But how to do this?

I know this is not related to mongoose.ws, but maybe someone help me with this, thank you.

**Comment by scaprile** [comment, marked as answer]
Many examples do that
Perhaps the timers example is the simplest: https://mongoose.ws/tutorials/timers/
https://mongoose.ws/tutorials
